test/test_autograd.py

```diff
@@ -2228,16 +2228,22 @@ def test_trapz(self):
                               True, f_args_variable, f_args_tensor)
 
     def test_cdist(self):
-        for p in [0, 1, 2, 3, 1.5, 2.5, float('inf')]:
-            f_args_variable = (torch.randn(S, S, requires_grad=True),
-                               torch.randn(S, S, requires_grad=True))
-
-            def f(a, b):
-                return torch.cdist(a, b, p)
-
-            f_args_tensor = deepcopy(unpack_variables(f_args_variable))
-            run_functional_checks(self, "test_cdist", "cdist", f,
-                                  True, f_args_variable, f_args_tensor)
+        def _test_cdist_for_size(sizes):
+            devices = ['cpu'] if not torch.cuda.is_available() else ['cpu', 'cuda']
```

Comment:
I know we use this idiom a lot but the better way to do this is
`devices = torch.testing.get_all_device_types()`